rust/src/ssh/detect.rs

```diff
@@ -90,3 +90,135 @@ pub extern "C" fn rs_ssh_tx_get_software(
 
     return 0;
 }
+
+#[no_mangle]
+pub extern "C" fn rs_ssh_tx_get_hassh(
+    tx: *mut std::os::raw::c_void,
+    buffer: *mut *const u8,
+    buffer_len: *mut u32,
+    direction: u8,
+) -> u8 {
+    let tx = cast_pointer!(tx, SSHTransaction);
+    match direction {
+        STREAM_TOSERVER => {
+            let m = &tx.cli_hdr.hassh;
+            if m.len() > 0 {
+                unsafe {
+                    *buffer = m.as_ptr();
+                }
+                unsafe {
+                    *buffer_len = m.len() as u32;
+                }
+                return 1;
+            }
+        }
+        _ => {}
+    }
+    unsafe {
+        *buffer = ptr::null();
+    }
+    unsafe {
+        *buffer_len = 0;
+    }
+
+    return 0;
+}
+
+#[no_mangle]
+pub extern "C" fn rs_ssh_tx_get_hassh_server(
+    tx: *mut std::os::raw::c_void,
```

Comment:
Why do we need two functions ? (When one parameter is `direction`)

Reply:
Now, there are two functions ```rs_ssh_tx_get_hassh``` (for hassh) and  ```rs_ssh_tx_get_hassh_string``` (for hassh string)

Reply:
I meant two functions like `rs_ssh_tx_get_hassh` and `rs_ssh_tx_get_hassh_server` when we could have one and use the direction

But then, with the md5 transform, we could have only one new keyword instead of 4 and still all the functionalities : https://suricata.readthedocs.io/en/suricata-5.0.3/rules/transforms.html#to-md5